### PR TITLE
Bumped Promises min version to 2.0.0 (#8373)

### PR DESCRIPTION
**Pull request by a-25**
Promises are of 2.0.0 minimum version now.

### CODE REVIEW

File: FirebaseAppCheck.podspec [hunk 1]
```diff
@@ -43,7 +43,7 @@ Pod::Spec.new do |s|
   s.tvos.weak_framework = 'DeviceCheck'
 
   s.dependency 'FirebaseCore', '~> 8.0'
-  s.dependency 'PromisesObjC', '>= 1.2', '< 3.0'
+  s.dependency 'PromisesObjC', '>= 2.0.0', '< 3.0'
```
Comment (paulb777) [inline FirebaseAppCheck.podspec:46]: We should use the standard CocoaPods `~` syntax:

s.dependency 'PromisesObjC', '~> 2.0'

**Review by paulb777 — COMMENTED** [review]
Thanks @a-25.  I'll mark this PR for our next major release.

**Comment by paulb777** [comment]
I created #9172 for the specs_checking CI issue

File: FirebaseAppCheck.podspec (hunk 1 above)
Comment (a-25) [reply FirebaseAppCheck.podspec]: Fixed

File: FirebaseAppCheck.podspec [hunk 2]
```diff
@@ -43,7 +43,7 @@ Pod::Spec.new do |s|
   s.tvos.weak_framework = 'DeviceCheck'
 
   s.dependency 'FirebaseCore', '~> 8.0'
-  s.dependency 'PromisesObjC', '>= 1.2', '< 3.0'
```
Comment (paulb777) [inline FirebaseAppCheck.podspec:46]: Should be 2.0 and not 2.0.0 to allow minor updates as well as patch updates. Also the '< 3.0' should be removed. Details about syntax at https://guides.cocoapods.org/syntax/podspec.html#dependency

Comment (a-25) [reply FirebaseAppCheck.podspec:46]: Fixed

**Comment by paulb777** [comment]
This PR could be rebased to v9. Otherwise, we'll merge it once we merge v9 to master.

**Review by ncooke3 — APPROVED** [review]

**Comment by ncooke3** [comment]
Thanks @a-25! Sorry for the random force pushes. I was trying to use [GitHub CLI](https://cli.github.com/) to do an interactive rebase on `v9` branch from this specific PR but it didn't work as I expected. I undid it to retain your authorship and pushed a commit to fix the merge conflicts 🙂

**Comment by a-25** [comment]
Hello, @ncooke3. That's ok, thank you.
Is it something I can do for this branch or it is successfully merged?

**Comment by ncooke3** [comment]
No further action is needed– this PR is merged and staged for the v9 release. Thanks again for this contribution!

**Pull request merged**